core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -34,7 +34,7 @@
  * using a {@link StaticTableOperations}. This way no Catalog related calls are needed when reading the table data after
  * deserialization.
  */
-abstract class BaseMetadataTable implements Table, HasTableOperations, Serializable {
+public abstract class BaseMetadataTable implements Table, HasTableOperations, Serializable {
```

Comment:
We could have a method in `MetadataTableUtils` for that but these two options are similar to me.
Since `BaseTable` is public, maybe it is fine to have `BaseMetadataTable` public too?